File: backend/src/main/java/bridge/glaucio/backend/resolver/TarefaQueryResolver.java
package bridge.glaucio.backend.resolver;

import java.util.List;

import javax.annotation.Nullable;

import lombok.RequiredArgsConstructor;

import org.springframework.stereotype.Component;


import bridge.glaucio.backend.model.Tarefa;
import bridge.glaucio.backend.service.TarefaService;
import graphql.kickstart.tools.GraphQLQueryResolver;

@Component
@RequiredArgsConstructor
public class TarefaQueryResolver implements GraphQLQueryResolver {

	private final TarefaService tarefaService;

	public List<Tarefa> tarefas() {
		return this.tarefaService.getTarefas();
	}
}

File: backend/src/main/java/bridge/glaucio/backend/resolver/TarefaQueryResolver.kt
package bridge.glaucio.backend.resolver;

import javax.annotation.Nullable;

import lombok.RequiredArgsConstructor;

import org.springframework.stereotype.Component;


import bridge.glaucio.backend.model.Tarefa;
import bridge.glaucio.backend.service.TarefaService;
import graphql.kickstart.tools.GraphQLQueryResolver;

@Component
@RequiredArgsConstructor
class TarefaQueryResolver(private val tarefaService: TarefaService) : GraphQLQueryResolver {

	fun tarefas(): List<Tarefa> {
		val tarefaList : List<Tarefa> = this.tarefaService.getTarefas().toList();
		return tarefaList;
	}
}
